packages/aws-cdk-lib/aws-rds/lib/instance.ts

```diff
@@ -130,6 +147,7 @@ export abstract class DatabaseInstanceBase extends Resource implements IDatabase
       public readonly instanceEndpoint = new Endpoint(attrs.instanceEndpointAddress, attrs.port);
       public readonly engine = attrs.engine;
       protected enableIamAuthentication = true;
+      public readonly instanceResourceId = attrs.instanceResourceId;
```

Comment:
This is a new, required property. Is it a breaking change for `fromDatabaseInstanceAttributes`?

Reply:
> This is a new, required property. Is it a breaking change for `fromDatabaseInstanceAttributes`?

Ahh the AWS CodeBuild report suggests it is:

```log
err  - IFACE aws-cdk-lib.aws_rds.DatabaseInstanceAttributes: newly required property 'instanceResourceId' added: input to aws-cdk-lib.aws_rds.DatabaseInstanceBase.fromDatabaseInstanceAttributes [strengthened:aws-cdk-lib.aws_rds.DatabaseInstanceAttributes]
Some packages seem to have undergone breaking API changes. Please avoid.
```

Advice on how to proceed welcomed 🙏🏽.

Reply:
I think you can change this to an optional property in `IDatabaseInstance` and `DatabaseInstanceBase` and `fromDatabaseInstanceAttributes`.

Reply:
> I think you can change this to an optional property in `IDatabaseInstance` and `DatabaseInstanceBase` and `fromDatabaseInstanceAttributes`.

Done in b612e7ec9ba873e5d652001c10e7bcbe3e7afdad. I'm not a huge fan of this as it means one has to ensure the property is not `undefined` when accessing it, which adds bloat IMO:

```ts
const db = new DatabaseInstance();
const resourceId: string | undefined = db.instanceResourceId;

if (!resourceId) {
  // unhappy path
}
else {
  // happy path
}
```

Alternatively, one can use [non-null assertion](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator), however I've worked in projects where this is yields warnings from the linter.

Additionally, this being optional makes it the [only optional property](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_rds.DatabaseInstance.html#properties). Not a problem in general, however might be an indication of not being desired?

Reply:
Yeah it's not great, but it's the only way to avoid breaking changes.